source/common/tcp_proxy/default_tcp_upstream_factory.cc

```diff
@@ -0,0 +1,65 @@
+#include "common/tcp_proxy/default_tcp_upstream_factory.h"
```

Comment:
This is still doing factories :-)

I'd like to start with the equivalent of https://github.com/envoyproxy/envoy/pull/11555/files
which is just cleaning up the APIs, commenting without switching over to factories.

Reply:
I think the factory simplifies Tcp::maybetunnel()

Will introduce this default factory in the next PR